packages/fetchai/skills/advanced_data_request/handlers.py

```diff
@@ -117,11 +126,12 @@ def _handle_response(self, http_msg: HttpMessage) -> None:
             json_path = output["json_path"]
 
             # find desired output data in msg_body
-            value = find(json_path, msg_body)
+            value = cast(SupportsFloat, find(json_path, msg_body))
 
             # if value is a numeric type, store it as fixed-point with number of decimals
-            if isinstance(value, (int, float)):
-                int_value = int(value * 10 ** model.decimals)
+            if is_number(value):
```

Comment:
The `int_value` here with associated `decimals` is effectively a fixed-point representation of the value, which is kind of necessary when publishing to the oracle smart contract. However, we could consider defaulting to float here (or high precision decimal) and only do the conversion before updating the contract.